docs/dev/callstacks.qmd

```diff
@@ -0,0 +1,413 @@
+    squash_per_depth && squash!(graph, nodes_per_depth, max_depth, squash_methods)
+
+    set_coordinates!(graph, nodes_per_depth, max_depth, plot_non_Ribasim)
+
+    files = sort(unique(graph[i].file for i in labels(graph) if graph[i].mod == :Ribasim))
+    colors = distinguishable_colors(length(files)+1)[end:-1:2]
+    color_dict = OrderedDict(zip(files, colors))
+
+    delete!(theme(nothing), :resolution) # Needed because of a refactor in Makie going from resolution to size
+    f = Figure(; size = size)
+    ax = Axis(f[1, 1]; xlabel = "depth", xticks = 0:max_depth)
+    plot_edges!(ax, graph, max_depth, nodes_per_depth)
+    plot_labels!(ax, graph, max_depth, color_dict)
+    hideydecorations!(ax)
+    !isnothing(xlims) && xlims!(ax, xlims...)
+
+    # Build legend
+    elements = LegendElement[MarkerElement(color = c, marker = :rect) for c in values(color_dict)]
+    descriptions = String.(files)
+
+    push!(elements, LineElement(color = :black, linestyle = :dash))
+    push!(descriptions, "between scripts")
+
+    push!(elements, LineElement(color = :black, linestyle = :solid))
+    push!(descriptions, "within a script")
+
+    Legend(
+        f[1,2],
+        elements,
+        descriptions
+    )
+
+    f
+end
+
+#############################################
+
+using Ribasim
+using Random
+
+Random.seed!(1);
+```
+
+The plots below show the call stack within the Julia core for several important entrypoints. The function names are colored by the script in which they are defined, and the lines between the function names have random colors to be able to differentiate between them. Solid lines refer to calls to functions defined in the same script, dashed ones to functions defined in a different script.
+
+Note that these graphs are obtained by dynamic analysis, i.e. by running parts of the code with specific inputs. This means that there can be unshown paths through the code that are not reached for these particular inputs.
+
+## Parameter initialization
+
+```{julia}
+# | code-fold: true
+using SQLite
+toml_path = normpath(@__DIR__, "../../generated_testmodels/basic_transient/ribasim.toml")
+config = Ribasim.Config(toml_path)
+db_path = Ribasim.database_path(config)
+db = SQLite.DB(db_path)
+
+graph, verts = tracecall((Ribasim,), Ribasim.Parameters, (db, config))
+close(db)
+
+plot_graph(
+    graph;
+    size = (2000, 1200),
+    squash_methods = [
+        :n_neighbor_bounds_flow,
+        :n_neighbor_bounds_control,
+        :sort_by_function,
+        :neighbortypes
+    ],
+    xlims = (-0.4, 5.6)
+)
+```
+
+## `water_balance!` (The ODE system right hand side function)
+
+```{julia}
+# | code-fold: true
+using OrdinaryDiffEqCore: get_du
+model = Ribasim.Model(toml_path)
+du = get_du(model.integrator)
+(; u, p, t) = model.integrator
+graph, verts = tracecall((Ribasim,), Ribasim.water_balance!, (du, u, p, t))
+plot_graph(graph, size = (1700, 1000), xlims = (-0.4, 4.5))
+```
+
+## Allocation initialization
```

Comment:
Can you provide a sentence or two on each figure? Ideally (only) on something that stands out, or is not immediately clear.